### ISSUE TITLE
Access to Generator and Discriminator Losses in BasicMolGAN

### ISSUE THREAD

**Issue by moben1** [deepchem/deepchem#3663]
## ❓ Questions & Help

**Context:**
I am currently working with the BasicMolGAN model in DeepChem and have encountered a limitation regarding the accessibility of training loss data for the generator and discriminator. As part of my project, I need to monitor and analyze the training progress of the GAN, specifically by plotting the generator and discriminator losses over epochs. However, the current implementation of fit_gan in BasicMolGAN does not seem to provide an easy way to access or record these loss values.

**Suggestion:**
This could be implemented in the fit_gan method, within the feed_dict loop for instance
```python
self.gen_losses.append(gen_error / max(1, gen_average_steps))
self.disc_losses.append(discrim_error / max(1, discrim_average_steps))
```

**Current alternative??**
In the meantime, could you suggest any current alternative methods or best practices within DeepChem to ensure that a GAN is well-trained or fitted? It is also to decide the parameters of the fit_gan and the BasicMolGan optimally.
Any guidance on how to monitor the training process effectively using the existing framework would be greatly appreciated.

Thank you for considering this request. I look forward to any suggestions or feedback from the DeepChem community.

**Comment by rbharath** [comment]
@shreyasvinaya Can you please take a look at this proposal and see if we can use callbacks to get this information?

@moben1 Thank you for the input! We will discuss and get back to you with some suggestions. We don't currently have a best practices guide, but we will be working on improvements to our tutorials and can add some best practices

**Comment by shreyasvinaya** [comment]
Hi @moben1 a workaround to this would be to set up the checkpoint interval to 1 and then parse the output that is printed out. that should get you what you require
